File: API/Entities/Post.cs
namespace API.Entities
{
    public class Post
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime DateCreated { get; set; }
        public bool HasPhoto { get; set; }
        public int LikesCount { get; set; }
        public int DislikesCount { get; set; }

        public string AppUserId { get; set; }
        public AppUser AppUser { get; set; }
        public PhotoPost PhotoPost { get; set; }
        public ICollection<Like> Likes { get; set; }
        public ICollection<PostComment> PostComments { get; set; }
    }
}

File: client/src/app/models/post.ts
import { Like } from "./like";

export class Post {
    id: number;
    title: string;
    description: string;
    dateCreated: Date;
    hasPhoto: boolean;
    photoUrl: string;
    username: string;
    userPhoto: string;
    likesCount: number;
    dislikesCount: number;
    likes: Like[] = [];
}